go/storage/filesystemdatabase.go

```diff
@@ -99,115 +91,35 @@ func (db *FilesystemDatabase) GetIssuerAndDatesFromCache() ([]IssuerDate, error)
 	return issuerList, nil
 }
 
-func (db *FilesystemDatabase) ListExpirationDates(aNotBefore time.Time) ([]ExpDate, error) {
-	return db.backend.ListExpirationDates(context.Background(), aNotBefore)
-}
-
-func (db *FilesystemDatabase) ListIssuersForExpirationDate(expDate ExpDate) ([]Issuer, error) {
-	return db.backend.ListIssuersForExpirationDate(context.Background(), expDate)
-}
-
 func (db *FilesystemDatabase) SaveLogState(aLogObj *CertificateLog) error {
-	ctx, ctxCancel := context.WithCancel(context.Background())
-	defer ctxCancel()
-	err := db.extCache.StoreLogState(aLogObj)
-	if err != nil {
-		glog.Warningf("Couldn't store log state for %s: %s", aLogObj, err)
-	}
-	return db.backend.StoreLogState(ctx, aLogObj)
+	return db.extCache.StoreLogState(aLogObj)
 }
 
 func (db *FilesystemDatabase) GetLogState(aUrl *url.URL) (*CertificateLog, error) {
-	ctx, ctxCancel := context.WithCancel(context.Background())
-	defer ctxCancel()
 	shortUrl := fmt.Sprintf("%s%s", aUrl.Host, aUrl.Path)
 
 	log, cacheErr := db.extCache.LoadLogState(shortUrl)
 	if log != nil {
 		return log, cacheErr
 	}
 
-	log, backendErr := db.backend.LoadLogState(ctx, shortUrl)
-	if log != nil {
-		return log, backendErr
-	}
-
-	glog.Warningf("Allocating brand new log for %+v, cache err=%v, backend err=%v", shortUrl, cacheErr, backendErr)
+	glog.Warningf("Allocating brand new log for %+v, cache err=%v", shortUrl, cacheErr)
 	return &CertificateLog{
 		ShortURL: shortUrl,
 	}, nil
 }
 
-func (db *FilesystemDatabase) markDirty(aExpiration *time.Time) error {
-	subdirName := aExpiration.Format(kExpirationFormat)
-	return db.backend.MarkDirty(subdirName)
-}
-
-func getSpki(aCert *x509.Certificate) SPKI {
-	if len(aCert.SubjectKeyId) < 8 {
-		digest := sha1.Sum(aCert.RawSubjectPublicKeyInfo)
-
-		glog.V(2).Infof("[issuer: %s] SPKI is short: %v, using %v instead.",
-			aCert.Issuer.String(), aCert.SubjectKeyId, digest[0:])
-		return SPKI{digest[0:]}
-	}
-
-	return SPKI{aCert.SubjectKeyId}
-}
-
 func (db *FilesystemDatabase) Store(aCert *x509.Certificate, aIssuer *x509.Certificate,
 	aLogURL string, aEntryId int64) error {
 	expDate := NewExpDateFromTime(aCert.NotAfter)
 	issuer := NewIssuer(aIssuer)
 	knownCerts := db.GetKnownCertificates(expDate, issuer)
 
-	ctx, ctxCancel := context.WithCancel(context.Background())
-	defer ctxCancel()
-
-	headers := make(map[string]string)
-	headers["Log"] = aLogURL
-	headers["Recorded-at"] = time.Now().Format(time.RFC3339)
-	headers["Entry-id"] = strconv.FormatInt(aEntryId, 10)
-	pemblock := pem.Block{
-		Type:    "CERTIFICATE",
-		Headers: headers,
-		Bytes:   aCert.Raw,
-	}
-
 	serialNum := NewSerial(aCert)
 
-	certWasUnknown, err := knownCerts.WasUnknown(serialNum)
-	if err != nil {
-		return err
-	}
-
-	if certWasUnknown {
-		issuerDateSeenBefore, err := db.GetIssuerMetadata(issuer).Accumulate(aCert)
```

Comment:
This call to Accumulate is responsible for storing CRL and Issuer records in Redis. I've re-introduced it in the latest commit on dev.